### PR TITLE
chore: Upgrade some dependencies

### PR DESCRIPTION
**Pull request by francislavoie**
Mainly for https://github.com/caddyserver/caddy/issues/6219#issuecomment-2038265390, but might as well upgrade some other things while we're at it

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -12,35 +12,34 @@ require (
...
 	github.com/stretchr/testify v1.9.0
 	github.com/tailscale/tscert v0.0.0-20230806124524-28a91b69a046
-	github.com/yuin/goldmark v1.5.6
+	github.com/yuin/goldmark v1.7.1
 	github.com/yuin/goldmark-highlighting/v2 v2.0.0-20230729083705-37449abec8cc
-	go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp v0.46.1
+	go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp v0.49.0
 	go.opentelemetry.io/contrib/propagators/autoprop v0.42.0
-	go.opentelemetry.io/otel v1.21.0
+	go.opentelemetry.io/otel v1.24.0
```
Comment (mohammed90) [inline go.mod:33]: We need to keep otel at 1.21.0, per Matt's comment here: https://github.com/caddyserver/caddy/pull/6198#pullrequestreview-1964063955

**Review by mohammed90 — CHANGES_REQUESTED** [review]
Thanks! You beat me :) One tiny thing

File: go.mod (hunk 1 above)
Comment (francislavoie) [reply go.mod:33]: Oops, forgot which ones we're locking. Thanks

Comment (mohammed90) [reply go.mod:33]: We probably should add comments around them or somewhere in the file

**Review by mohammed90 — APPROVED** [review]

**Comment by areriklq** [comment]
I am late here, but is there an official stable Docker image with these changes?
2.7.6 was released before this change was merged.

**Comment by mohammed90** [comment]
> I am late here, but is there an official stable Docker image with these changes? 2.7.6 was released before this change was merged.

Not yet. This PR will be part of the v2.8.0 release, which is yet to take place. The rc/beta will released soon followed by the official release of v2.8.0, which will eventually be published in Docker Hub. If you cannot wait, you can build a custom Docker image with Caddy built from source at `master` or the latest git commit hash. Build instructions are on our [website](https://caddyserver.com/docs/build#docker) here.

**Comment by areriklq** [comment]
> > I am late here, but is there an official stable Docker image with these changes? 2.7.6 was released before this change was merged.
> 
> Not yet. This PR will be part of the v2.8.0 release, which is yet to take place. The rc/beta will released soon followed by the official release of v2.8.0, which will eventually be published in Docker Hub. If you cannot wait, you can build a custom Docker image with Caddy built from source at `master` or the latest git commit hash. Build instructions are on our [website](https://caddyserver.com/docs/build#docker) here.

Thanks for the clarification! 👍 I'll have to consider my options then.

**Pull request merged**